internal/exec/exec.go

```diff
@@ -208,8 +209,48 @@ func execFieldSelection(ctx context.Context, r *Request, s *resolvable.Schema, f
 			if f.field.ArgsPacker != nil {
 				in = append(in, f.field.PackedArgs)
 			}
+
+			// Before hook directive visitor
+			if len(f.field.Directives) > 0 {
+				for _, directive := range f.field.Directives {
+					if visitor, ok := r.Visitors[directive.Name.Name]; ok {
+						values := make([]interface{}, 0, len(in))
+						for _, inValue := range in {
+							values = append(values, inValue.Interface())
+						}
+
+						visitorErr := visitor.Before(ctx, directive, values)
+						if visitorErr != nil {
+							err := errors.Errorf("%s", visitorErr)
+							err.Path = path.toSlice()
+							err.ResolverError = visitorErr
+							return err
+						}
+					}
+				}
+			}
+
+			// Call method
 			callOut := res.Method(f.field.MethodIndex).Call(in)
 			result = callOut[0]
+
+			// After hook directive visitor (when no error is returned from resolver)
+			if !f.field.HasError && len(f.field.Directives) > 0 {
+				for _, directive := range f.field.Directives {
+					if visitor, ok := r.Visitors[directive.Name.Name]; ok {
+						returned, visitorErr := visitor.After(ctx, directive, result.Interface())
+						if visitorErr != nil {
+							err := errors.Errorf("%s", visitorErr)
+							err.Path = path.toSlice()
+							err.ResolverError = visitorErr
+							return err
+						} else {
+							result = reflect.ValueOf(returned)
+						}
```

Comment:
```suggestion
						modified, visitorErr := visitor.After(ctx, directive, result.Interface())
						if visitorErr != nil {
							err := errors.Errorf("%s", visitorErr)
							err.Path = path.toSlice()
							err.ResolverError = visitorErr
							return err
						} else {
							result = reflect.ValueOf(modified)
						}
```

I'd prefer using the name `modified` to show the next person who reads this code that the directives can actually override the returned value.